ipc/provider/src/lib.rs

```diff
@@ -572,6 +598,25 @@ impl IpcProvider {
 
         conn.manager().get_chain_id().await
     }
+
+    pub async fn get_chain_head_height(&self, subnet: &SubnetID) -> anyhow::Result<ChainEpoch> {
+        let conn = match self.connection(subnet) {
+            None => return Err(anyhow!("target subnet not found")),
+            Some(conn) => conn,
+        };
+
+        conn.manager().chain_head_height().await
+    }
+
+    pub async fn get_genesis_epoch(&self, subnet: &SubnetID) -> anyhow::Result<ChainEpoch> {
+        let parent = subnet.parent().ok_or_else(|| anyhow!("no parent found"))?;
+        let conn = match self.connection(&parent) {
+            None => return Err(anyhow!("parent subnet not found")),
+            Some(conn) => conn,
+        };
+
+        conn.manager().chain_head_height().await
```

Comment:
Also, this doesn't seem what you want to do, right? :)

Reply:
right, this method should be removed.